Code/GraphMol/SmilesParse/SmartsWrite.cpp

```diff
@@ -56,7 +56,8 @@ std::string _combineChildSmarts(std::string cs1, unsigned int features1,
           "binary tree");
     }
     res += cs1;
-    res += ",";
+    if (!(cs1.empty() || cs2.empty()))
```

Comment:
What happens if cs1 is empty and cs2 is not, doesn't this make

",<cs2>"?

Which looks wrong to me ...

Reply:
@bp-kelley No, it doesn't, unless I'm having a bad day (which may well be :-) ):
```
class std_str(str):
    def empty(self):
        return (len(self) == 0)

cs1 = std_str("")
cs2 = std_str("@")
res = std_str("")

res += cs1
if (not (cs1.empty() or cs2.empty())):
    res += ","
res += cs2

res
'@'

cs1 = std_str("@")
cs2 = std_str("")
res = std_str("")

res += cs1
if (not (cs1.empty() or cs2.empty())):
    res += ","
res += cs2

res
'@'

cs1 = std_str("-")
cs2 = std_str("@")
res = std_str("")

res += cs1
if (not (cs1.empty() or cs2.empty())):
    res += ","
res += cs2

res
'-,@'
```

Reply:
For some reason the if !(a|b) is messing with my head.  I find it easier to understand:

```
if ( !cs1.empty() and !cs2.empty()) 
```